emily_cron/app/clients/mempool.py

```diff
@@ -7,11 +7,48 @@
 logger = logging.getLogger(__name__)
 
 
+def _collect_rbf_txids(data: dict[str, Any] | None) -> list[str]:
+    """Recursively collect all RBF transaction IDs from the replacement chain.
+
+    Args:
+        data: Transaction replacement data from mempool API
+
+    Returns:
+        List of transaction IDs that replaced the original transaction
+    """
+    txids = []
+
+    if data is None:
+        return txids
+
+    if tx := data.get("tx", {}):
+        if txid := tx.get("txid"):
+            txids.append(txid)
+
+    for replacement in data.get("replaces", []):
+        txids.extend(_collect_rbf_txids(replacement))
+
+    return txids
+
+
 class MempoolAPI(APIClient):
     """Client for interacting with the Mempool API."""
 
     BASE_URL = settings.MEMPOOL_API_URL
 
+    @classmethod
+    def check_for_rbf(cls, txid: str) -> list[str]:
+        """Check if a Bitcoin transaction was replaced by RBF.
+
+        Args:
+            txid: The transaction ID to check
+
+        Returns:
+            List of transaction IDs that replaced the original transaction
```

Comment:
nit: _[txid] that replaced the original transaction_ here is a bit confusing, IIUC `_collect_rbf_txids` returns all the txs in the replacement "tree", where the `txid` argument passed in here could be in any point of that tree, so we get both the txs replaced by `txid` as well as the ones replacing `txid` (and even eventual siblings that are not either of those). A similar nit does applies to the doc of `_collect_rbf_txids` and `rbf_txids` comment.